examples/session-management/main.bal

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/uuid;
+import ballerinax/redis;
+
+// Define the Redis client configuration
+configurable string host = ?;
+configurable int port = ?;
+
+redis:ConnectionConfig redisConfig = {
+    connection: {
+        host,
+        port
+    }
+};
+
+// Initialize the Redis client
+redis:Client redis = check new (redisConfig);
+
+// Define the HTTP service
+service / on new http:Listener(9090) {
+
+    // Resource to handle login and session creation
+    resource function post loginAndCreateSession(http:Request req) returns string|http:Unauthorized|error {
+        // Extract username and password from the request
+        string username = (check req.getFormParams()).get("username");
+        string password = (check req.getFormParams()).get("password");
```

Comment:
Rather than doing this, shall we include basic-auth with file user-store [1]

[1] - [https://ballerina.io/learn/by-example/http-service-basic-authentication-file-user-store/](https://ballerina.io/learn/by-example/http-service-basic-authentication-file-user-store)

Reply:
+1 Updated with [Update session management sample with user-store based auth](https://github.com/ballerina-platform/module-ballerinax-redis/pull/220/commits/686a40b0e4c355f050cad791bb43cc9cfc03d970)